polaris-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -60,6 +61,13 @@
       &::before {
         background: var(--p-color-border-disabled);
       }
+      #{$se23} & {
+        background-color: var(--p-color-bg-disabled);
+        // stylelint-disable-next-line max-nesting-depth, selector-max-combinators -- pse23 override
```

Comment:
Didn't notice this in the initial pass, pre-uplift we use `color-border-disabled` for both background and border colour.